airflow/metrics/datadog_logger.py

```diff
@@ -51,7 +55,33 @@
     )
 
 
```

Comment:
Nice work seeing and extracting the common code here, it really helps clean up the whole class.   If you are making this and statsd both inherit from StatsLogger, could you move this decorator into the base class for both to use, or am I missing some difference between them?

Reply:
There are two differences mainly that's why I didn't do that:

1. Tags Format:
prepare_stat_with_tags converts tags into a tags_list (e.g., ["key:value"]) for use in kwargs.
prepare_metric_name_with_tags keeps tags as a dictionary, formatted for InfluxDB.

2. Return Behavior:
prepare_stat_with_tags runs the function directly with tags_list in kwargs.
prepare_metric_name_with_tags returns a modified metric_name with tags if influxdb_tags_enabled is True.

Reply:
Sounds good then.